service/storage_service.cc

```diff
@@ -2665,6 +2741,11 @@ future<> topology_coordinator::run() {
         _topo_sm.event.broadcast();
     });
 
+    _load_stats_refresh_trigger.arm_periodic(load_stats_refresh_interval);
+    auto disarm_trigger = seastar::defer([&] {
+        _load_stats_refresh_trigger.cancel();
+    });
```

Comment:
I don't see why we have both the timer and the explicit refresh in maybe_start_tablet_migration().

Reply:
Yes but why do anything in maybe_start_tablet_migration()? You have statistics refreshed periodically, why add another trigger?

Reply:
Relies on a split completed event triggering rebalancing, but that's seen by the load balancer, no?

Reply:
> Yes but why do anything in maybe_start_tablet_migration()? You have statistics refreshed periodically, why add another trigger?

Tomek proposed reusing the same path, where resize decision and migration could be encoded in the same migration plan by the balancer. And the idea is that resize and migration can happen concurrently. Updates to tablet metadata would happen in the same update path.

If we want to decouple them, we can restrict future<migration_plan> tablet_allocator::balance_tablets() to migration and introduce a future<resize_plan> tablet_allocator::resize_tablets() that is invoked after we pull stats in the RPC.

Reply:
> Relies on a split completed event triggering rebalancing, but that's seen by the load balancer, no?

The timer is only to awake the topology event handler, which calls maybe_start_tablet_migration(). I think the right direction is to have an event loop for pulling stats which sleeps until the next table times out. So statistics can be pulled independent of the migration path.

Reply:
Sure it should be the same path. I just don't see why the timer needs to be managed in two places.

I don't understand what "next table times out" means.

The load balancer shouldn't rely on time, just on events (some events can be time driven like statistics collection, but the loop need not know that).

Reply:
I think the confusion is that I was explaining the motivation for the current state. Yes, no need for timer to be handled in more than one place. 

There will be a loop collecting stats every at least N seconds (the minimum frequency among all tables), and on completion, it wakes up the event handler (that in turn invokes the load balancer). When I said time out, I was referring to the expiration time for a table to have stats refreshed on its behalf.

I think we're on the same page.